### PR TITLE
 tests: internal: stream_processor: fix when test path is too long.

### PR DESCRIPTION
**Pull request by pwhelan**
# Summary

Fix the usage of datafile so it will not overflow when testing from a long repository path. This patch uses snprintf as well as defining the size of the string with PATH_MAX.

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:
- [N/A] Example configuration file for the change
- [N/A] Debug log output from testing the change

- [N/A] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

If this is a change to packaging of containers or native binaries then please confirm it works for all targets.
- [N/A] Run [local packaging test](./packaging/local-build-all.sh) showing all targets (including any new ones) build.
- [N/A] Set `ok-package-test` label to test for all targets (requires maintainer to do).

**Documentation**

- [N/A] Documentation required for this feature

**Backporting**

- [ ] Backport to latest stable release.

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

File: tests/internal/stream_processor.c [hunk 1]
```diff
@@ -529,8 +529,13 @@ static void test_window()
             task->window.fd_hop = 1;
             double record_timestamp = 1.0;
             for (t = 0; t < check->window_size_sec + check->window_hop_sec; t++) {
-                sprintf(datafile, "%s%d.mp",
+                ret = snprintf(datafile, sizeof(datafile)-1, "%s%d.mp",
                         DATA_SAMPLES_HOPPING_WINDOW_PATH, t + 1);
+                TEST_CHECK(ret <= sizeof(datafile)-1);
+
+                if (ret > sizeof(datafile)-1) {
```
Comment (nokute78) [inline tests/internal/stream_processor.c:536]: These lines can be replaced like 
```c
if (!TEST_CHECK(ret <= sizeof(datafile)-1)) {
```

File: tests/internal/stream_processor.c [hunk 2]
```diff
@@ -666,8 +671,14 @@ static void test_snapshot()
 
         /* Read 1.mp -> 5.mp message pack buffers created for window tests */
         for (t = 0; t < 5; t++) {
-            sprintf(datafile, "%s%d.mp",
+
+            ret = snprintf(datafile, sizeof(datafile)-1, "%s%d.mp",
                     DATA_SAMPLES_HOPPING_WINDOW_PATH, t + 1);
+            TEST_CHECK(ret <= sizeof(datafile)-1);
+
+            if (ret > sizeof(datafile)-1) {
```
Comment (nokute78) [inline tests/internal/stream_processor.c:679]: Ditto

**Review by nokute78 — CHANGES_REQUESTED** [review]
Could you check review comments ?

**Comment by edsiper** [comment]
@pwhelan please fix the commit prefix, it should be `tests: internal: stream_processor: ...`

**Comment by pwhelan** [comment]
> tests: internal: stream_processor

As requested.

**Review by nokute78 — APPROVED** [review]
LGTM.

**Pull request merged**